src/Build/BackEnd/Components/Communications/NodeProviderOutOfProc.cs

```diff
@@ -72,17 +72,22 @@ public int AvailableNodes
         /// Derived from the binary timestamp to avoid mixing binary versions,
         /// Is64BitProcess to avoid mixing bitness, and enableNodeReuse to
         /// ensure that a /nr:false build doesn't reuse clients left over from
-        /// a prior /nr:true build.
+        /// a prior /nr:true build. The enableLowPriority flag is to ensure that
+        /// a build with /low:false doesn't reuse clients left over for a prior
+        /// /low:true build.
         /// </summary>
         /// <param name="enableNodeReuse">Is reuse of build nodes allowed?</param>
-        internal static long GetHostHandshake(bool enableNodeReuse)
+        /// <param name="enableLowPriority">Is the build running at low priority?</param>
+        internal static long GetHostHandshake(bool enableNodeReuse, bool enableLowPriority)
         {
             long baseHandshake = Constants.AssemblyTimestamp;
 
             baseHandshake = baseHandshake*17 + EnvironmentUtilities.Is64BitProcess.GetHashCode();
 
             baseHandshake = baseHandshake*17 + enableNodeReuse.GetHashCode();
 
+            baseHandshake = baseHandshake*17 + enableLowPriority.GetHashCode();
```

Comment:
Question: would it be possible to skip the plumbing of the `lowPriority` flag down to this point and look instead at the current process's priority? That might help with super-advanced cases where a user explicitly sets the entry-point MSBuild to some priority int but doesn't explicitly use `/low`. And it'd simplify the code around here.

Reply:
Before deciding on this please read my other comment.